File: Gifting.Public/src/app/models/idea.ts
import { Injectable } from '@angular/core';

@Injectable()
export class Idea {
	constructor(
		public name: string,
		public description?: string,
		public imageUrl?: string,
		public whereToBuy?: string,
		public occasionId?: number,
		public granteeId?: number,
		public id?: number,
		public dateCreated?: Date
	) {
	}
}

File: Gifting.Models/Entities/Idea.cs
using System;

namespace Gifting.Models.Entities
{
	public class Idea
	{
		public long Id { get; set; }

		public string Name { get; set; }

		public string Description { get; set; }

		public string ImageUrl { get; set; }

		public string WhereToBuy { get; set; }

		public long? OccasionId { get; set; }

		public long? GranteeId { get; set; }

		public DateTime DateCreated { get; set; }

		public string Tags { get; set; }

		public long? UserId { get; set; }

		public bool IsApproved { get; set; }
	}
}